packages/hooks/src/useDocumentEventListener.ts

```diff
@@ -28,10 +30,13 @@ export default function useDocumentEventListener(
 
     const callback = e => memoizedEventCallback.current(e);
 
-    document.addEventListener(type, callback, options);
+    element.addEventListener(type, callback, options);
 
     return () => {
       document.removeEventListener(type, eventCallback);
     };
-  }, [enabled, ...(dependencies ? dependencies : [])]);
+  }, [
+    enabled,
+    ...(dependencies ? dependencies : [type, eventCallback, options]),
```

Comment:
Let's have dependencies in the argument just be default to this: `useEventListener(type, callback, dependencies = [type, eventCallback, options])`.

Reply:
👍